Tools/scripts/win_add2path.py

```diff
@@ -11,43 +11,67 @@
 import site
 import os
 import winreg
+import ctypes
 
 HKCU = winreg.HKEY_CURRENT_USER
 ENV = "Environment"
 PATH = "PATH"
-DEFAULT = "%PATH%"
 
 def modify():
     pythonpath = os.path.dirname(os.path.normpath(sys.executable))
     scripts = os.path.join(pythonpath, "Scripts")
     appdata = os.environ["APPDATA"]
-    if hasattr(site, "USER_SITE"):
-        usersite = site.USER_SITE.replace(appdata, "%APPDATA%")
-        userpath = os.path.dirname(usersite)
-        userscripts = os.path.join(userpath, "Scripts")
-    else:
-        userscripts = None
 
     with winreg.CreateKey(HKCU, ENV) as key:
         try:
-            envpath = winreg.QueryValueEx(key, PATH)[0]
-        except OSError:
-            envpath = DEFAULT
+            envpath, dtype = winreg.QueryValueEx(key, PATH)
+        except FileNotFoundError:
+            envpath, dtype = "", winreg.REG_EXPAND_SZ
+            pass
+        except:
+            raise OSError("Failed to load PATH value")
+
+        if hasattr(site, "USER_SITE") and dtype == winreg.REG_EXPAND_SZ:
```

Comment:
This `hasattr` check doesn't apply to the subsequent `elif` block, or rather, if the first check here fails then we may end up crashing in the elif block.